azurerm/resource_arm_private_endpoint.go

```diff
@@ -0,0 +1,380 @@
+			d.Set("subnet_id", subnet.ID)
+		}
+	}
+	tags.FlattenAndSet(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmPrivateEndpointDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).network.PrivateEndpointClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["privateEndpoints"]
+
+	future, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if response.WasNotFound(future.Response()) {
+			return nil
+		}
+		return fmt.Errorf("Error deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	return nil
+}
+
+func expandArmPrivateEndpointPrivateLinkServiceConnection(input []interface{}) *[]network.PrivateLinkServiceConnection {
+	results := make([]network.PrivateLinkServiceConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		privateLinkServiceID := v["private_link_service_id"].(string)
+		groupIds := v["group_ids"].([]interface{})
+		requestMessage := v["request_message"].(string)
+		name := v["name"].(string)
+
+		result := network.PrivateLinkServiceConnection{
+			Name: utils.String(name),
+			PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+				GroupIds:             utils.ExpandStringSlice(groupIds),
+				PrivateLinkServiceID: utils.String(privateLinkServiceID),
+				RequestMessage:       utils.String(requestMessage),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func flattenArmPrivateEndpointPrivateLinkServiceConnection(input *[]network.PrivateLinkServiceConnection) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		v := make(map[string]interface{})
+
+		if name := item.Name; name != nil {
+			v["name"] = *name
+		}
+		if privateLinkServiceConnectionProperties := item.PrivateLinkServiceConnectionProperties; privateLinkServiceConnectionProperties != nil {
+			v["group_ids"] = utils.FlattenStringSlice(privateLinkServiceConnectionProperties.GroupIds)
+			v["private_link_service_connection_state"] = flattenArmPrivateLinkServiceConnectionState(privateLinkServiceConnectionProperties.PrivateLinkServiceConnectionState)
+			if privateLinkServiceId := privateLinkServiceConnectionProperties.PrivateLinkServiceID; privateLinkServiceId != nil {
+				v["private_link_service_id"] = *privateLinkServiceId
+			}
+			if requestMessage := privateLinkServiceConnectionProperties.RequestMessage; requestMessage != nil {
+				v["request_message"] = *requestMessage
+			}
+		}
```

Comment:
This could would be easier to review & read if
```suggestion
		if props := item.PrivateLinkServiceConnectionProperties; props != nil {
```

Reply:
Fixed.